File: darkrodry/go/kata/game_test.go
package kata

import "testing"

var game Game

func setUp() {
	game = Game{}
}

func rollMany(n, pins int) {
	for i := 0; i < n; i++ {
		game.roll(pins)
	}
}

func rollSpare() {
	game.roll(5)
	game.roll(5)
}

func assertScore(expected, score int, t *testing.T) {
	if score != expected {
		t.Errorf("Game.score() expected %d, got %d", expected, score)
	}
}

func TestGutterGame(t *testing.T) {
	setUp()
	rollMany(20, 0)
	assertScore(0, game.score(), t)
}

func TestAllOnes(t *testing.T) {
	setUp()
	rollMany(20, 1)
	assertScore(20, game.score(), t)
}

func TestOneSpare(t *testing.T) {
	setUp()
	rollSpare()
	game.roll(3)
	rollMany(17, 0)
	assertScore(16, game.score(), t)
}

func TestOneStrike(t *testing.T) {
	setUp()
	game.roll(10)
	game.roll(4)
	game.roll(3)
	rollMany(16, 0)
	assertScore(24, game.score(), t)
}

func TestPerfectGame(t *testing.T) {
	setUp()
	rollMany(12, 10)
	assertScore(300, game.score(), t)
}

File: darkrodry/ruby/kata/game_test.rb
require 'test/unit'
require_relative 'game' 

class GameTest < Test::Unit::TestCase

  def setup
    @game = Game.new
  end

  def roll_many(n, pins)
    n.times do
      @game.roll(pins)
    end
  end

  def test_gutter_game
    setup
    roll_many(20, 0)
    assert_equal 0, @game.score
  end

  def test_all_ones
    setup
    roll_many(20, 1)
    assert_equal 20, @game.score
  end

  def test_one_spare
    setup
    @game.roll(5)
    @game.roll(5)
    @game.roll(3)
    roll_many(17, 0)
    assert_equal 16, @game.score
  end
  
  def test_one_strike
    setup
    @game.roll(10)
    @game.roll(3)
    @game.roll(4)
    roll_many(16, 0)
    assert_equal 24, @game.score
  end

  def test_perfect_game
    setup
    roll_many(12, 10)
    assert_equal 300, @game.score
  end

end
